Fix: Update CI Job names with required config params

karthiknathan06 opened the pull request:
## Description
This PR updates the GitHub Actions workflow for python pub sub test to improve debugging by printing the full matrix configuration in the logs for each job. 

The previous job name was truncated, making it hard to track matrix parameters used in each job. This change ensures that the full matrix configuration is printed to the logs.

## Fixes
- Added a step to output the full matrix configuration to the logs using `toJson(matrix)`.
- Trimmed the job header with Language Version, Engine version and TARGET
- Above changes are done for Python, node, Go, CSharp, Java WF

### Issue link

This Pull Request is linked to issue ([URL](https://github.com/valkey-io/valkey-glide/issues/3398)): [3398]

## Notes for Reviewers
- Please review if the updated example aligns with the requirement.
- Let me know if any additional clarifications are needed.

### Checklist

Before submitting the PR make sure the following are checked:

-   [Yes] This Pull Request is related to one issue.
-   [Yes] Commit message has a detailed description of what changed and why.
-   [NA] Tests are added or updated.
-   [NA] CHANGELOG.md and documentation files are updated.
-   [Yes] Destination branch is correct - main or release
-   [Yes] Create merge commit if merging release branch into main, squash otherwise.

karthiknathan06 commented:
<img width="1072" alt="Screenshot 2025-03-22 at 1 29 36 PM" src="https://github.com/user-attachments/assets/20f26c4f-03e5-4d85-9fb0-ae73e1969a21" />

Added all matrix params in logs.

avifenesh commented:
@karthiknathan06 Thanks!
Some misunderstanding - the issue is for all the matrix WF, both py PubSub, but also python, node, java, go and possibly C#,
And we would really like to have the title shorter to include the necessary parameters in a way that it can be seen, without digging into the logs. You think it is possible to shorten the title so it will all be seen?
The parameters we need are language version, engine version, and TARGET.
See the different parameters exist, and the ones we don't need: https://github.com/valkey-io/valkey-glide/tree/main/.github/json_matrices

avifenesh commented:
Note that you should sign the commits

karthiknathan06 commented:
Hi @avifenesh 
I have signed off the commits and also squashed commits. Also addressed below cases
1) trimmed header with required details.
2) Added extended matrix details in logs
3) extended these cases for all other supported languages

avifenesh commented:
@karthiknathan06 Amazing! That's great.
But is it possible to have from this: `[NodeTest(Node-16.x,EngineVersion-6.2,Target-x86_64-unknown-linux-gnu)]` to 
`[Node Tests - 16.x, Engine Version - 6.2, Target - x86_64-unknown-linux-gnu]` 
I don't think we need it to be code style, better having readable and easy to work with.

karthiknathan06 commented:
@avifenesh Thanks for the feedback. I have updated as per the suggestion.

avifenesh commented:
Perfect! @karthiknathan06 Thanks a lot. Im approving it, let me know if i can merge.

karthiknathan06 commented:
Thanks! Please go ahead with the merge @avifenesh
